tests/ops/qubit/test_state_prep.py

```diff
@@ -98,6 +98,61 @@ def test_QubitStateVector_broadcasting(self):
 class TestStateVector:
     """Test the state_vector() method of various state-prep operations."""
 
+    @pytest.mark.parametrize(
+        "num_wires,wire_order,one_position",
+        [
+            (2, None, (1, 0)),
+            (2, [1, 2], (1, 0)),
+            (3, [0, 1, 2], (0, 1, 0)),
+            (3, ["a", 1, 2], (0, 1, 0)),
+            (3, [1, 2, 0], (1, 0, 0)),
+            (3, [1, 2, "a"], (1, 0, 0)),
+            (3, [2, 1, 0], (0, 1, 0)),
+        ],
```

Comment:
Can we add a example with many reordering necessary?

Reply:
I did this below in a separate test (`test_QubitStateVector_reordering`) because the $\ket{01}$ state used in this test is a little trivial for convincing the reader of my point. lmk if you'd like more/different testing for it